nussl/core/audio_signal.py

```diff
@@ -1653,73 +1654,99 @@ def embed_audio(self, ext='.mp3', display=True):
+        in the order that they were applied.
 
         Args:
-            reset (bool): If True, clears out all effects in effect chains. Default=True
-            overwrite (bool): If True, overwrites existing audio_data in AudioSignal. Defult=False
-                Also clears out stft_data
+            reset (bool): If True, clears out all effects in effect chains following applying the effects. 
+                Default=True
+            overwrite (bool): If True, overwrites existing audio_data in AudioSignal. Default=False
+                Also clears out stft_data.
         Returns:
             self or new_signal (AudioSignal): If overwrite=True, then returns initially AudioSignal 
             with edited audio_data. Otherwise, returns a new AudioSignal new_signal.
         """
-        if not self._ffmpeg_effects_chain and not self._sox_effects_chain:
-            raise RuntimeError("No effect hooks have been called on this AudioSignal")
 
         new_signal = self
         if self._sox_effects_chain:
-            new_signal = effects.build_effects_sox(new_signal, self._sox_effects_chain)
+            new_signal = effects.apply_effects_sox(new_signal, self._sox_effects_chain)
         if self._ffmpeg_effects_chain:
-            new_signal = effects.build_effects_ffmpeg(new_signal, self._ffmpeg_effects_chain)
+            new_signal = effects.apply_effects_ffmpeg(new_signal, self._ffmpeg_effects_chain)
 
         if reset:
             self.reset_effects_chain()
         if overwrite:
             self.audio_data = new_signal.audio_data
+            self.effects_applied += new_signal.effects_applied
             self.stft_data = None
             return self
 
         return new_signal
         
-    def make_effect(self, effect, **params):
+    def make_effect(self, effect, **kwargs):
         """
-        Calls another effect hook.
+        Calls another effect hook. An example:
+
+        >>> signal.time_stretch(1.5)
+
+        Is the same as 
+        >>> signal.make_effect("time_stretch", factor=1.5)
+
         Args:
             effect (str): Function name of desired effect hook of the AudioSignal
-            **params: Additional parameters for given effect. 
+            **kwargs: Additional parameters for given effect. 
         Return:
             self: Inital AudioSignal with updated effect chains
         """
         
         try:
             effect_hook = getattr(self, effect, None)
-            effect_hook(**params)
+            effect_hook(**kwargs)
         except Exception:
-            raise AudioSignalException(f"Effect {effect} not found with parameters {params}")
+            raise AudioSignalException(f"Effect {effect} not found with parameters {kwargs}")
         
         return self
 
     def reset_effects_chain(self):
         """
-        Clears effects chain of AudioSignal
+        Clears effects chains of AudioSignal
```

Comment:
Add a line like "for more information on applying effects in AudioSignal see link:`apply_effects()`